src/PythonProjectBootstrapper/EntryPoint.py

```diff
@@ -177,21 +191,345 @@ def _ExecuteOutputDir(
+
+    # remove files no longer in template if they are unchanged
+    for removed_file_rel_path in removed_template_files:
+        removed_full_path = output_dir / removed_file_rel_path
+
+        if removed_full_path.is_file():
+            with open(removed_full_path, "rb") as removed_file:
+                current_hash = hashlib.file_digest(removed_file, hashlib.sha256).hexdigest()
+
+            original_hash = existing_manifest_dict[removed_file_rel_path]
+
+            if current_hash == original_hash:
+                removed_full_path.unlink()
+
+
 # ----------------------------------------------------------------------
+# Copies all generated files into the output directory and handles the creation/updating of the manifest file
+
+
+def _CopyToOutputDir(tmp_dir: Path, output_dir: Path) -> None:
+    PathEx.EnsureDir(tmp_dir)
+    PathEx.EnsureDir(output_dir)
+
+    # existing_manifest will be populated/updated as necessary and saved
+    generated_manifest: dict[Path, str] = _CreateManifest(tmp_dir)
+    existing_manifest: dict[Path, str] = {}
+
+    potential_manifest: Path = output_dir / ".manifest.yml"
+
+    # if this is not our first time generating, remove unwanted template files
+    if potential_manifest.is_file():
+        with open(potential_manifest, "r") as existing_manifest_file:
+            existing_manifest = yaml.load(existing_manifest_file, Loader=yaml.Loader)
+
+        _ConditionallyRemoveUnchangedTemplateFiles(
+            new_manifest_dict=generated_manifest,
+            existing_manifest_dict=existing_manifest,
+            output_dir=output_dir,
+        )
+
+    merged_manifest = dict(generated_manifest)
+    merged_manifest.update(existing_manifest)
+
+    # Ask user if they would like to overwrite their changes if any conflicts detected
+    for rel_filepath, generated_hash in generated_manifest.items():
+        output_dir_filepath: Path = output_dir / rel_filepath
+
+        if output_dir_filepath.is_file():
+            with open(output_dir_filepath, "rb") as current_file:
+                current_file_hash: str = hashlib.file_digest(
+                    current_file, hashlib.sha256
+                ).hexdigest()
+
+            # Changes detected in file and file modified by user (changes do not stem only from changes in the contents of the template file)
+            if (
+                generated_hash != current_file_hash
+                and current_file_hash != merged_manifest[rel_filepath]
+            ):
+
+                while True:
+                    sys.stdout.write(
+                        f"\nWould you like to overwrite your changes in {str(output_dir_filepath)}? [yes/no]: "
+                    )
+                    overwrite = input().strip().lower()
+
+                    if overwrite in ["yes", "y"]:
+                        break
+
+                    if overwrite in ["no", "n"]:
+                        shutil.copy2(output_dir_filepath, tmp_dir / rel_filepath)

```

Comment:
This copy is a bit surprising to me. Are we copying the existing file to the `tmp_dir` to make it easier to copy content from the `tmp_dir` back to the output dir? If so, that doesn't feel quite right to me. If this isn't what is happening, add a comment to explain what is going on.